api/go.mod

```diff
@@ -5,6 +5,7 @@ go 1.22.6
 require (
 	k8s.io/api v0.29.0
 	k8s.io/apimachinery v0.29.0
+	k8s.io/component-base v0.28.3
```

Comment:
This is already [fixed](https://github.com/RamenDR/ramen/blob/32d7d5e726f7ecb657e4fafecfc748e526f68a4c/go.mod#L32) in main.

Reply:
Not in the api/go.mod file

Reply:
```
$ grep component-base go.mod
        k8s.io/component-base v0.31.0
$ grep component-base api/go.mod
        k8s.io/component-base v0.28.3 // indirect
```

Reply:
Ah, missed that. LGTM.

I wonder why this missed the github action to catch the same, and also, we need to move API go deps higher, as well correct?